action.yml

```diff
@@ -46,24 +49,30 @@ runs:
         $pkg_list = $pkg_list | % { $_.Trim() }
         $pkg_list = $pkg_list | % { $_.Trim(',') }
 
+        $args = @(
+         '-qgnO',
+         '-l', 'C:\cygwin-packages',
+         '-R', '${{ inputs.install-dir }}'
+        )
+
         # default site if not specified
-        if (! '${{ inputs.site }}' ) {
+        if (! '${{ inputs.sites }}' ) {
+          $args += '-s'
           if ($platform -eq 'x86') {
-            $site = 'http://mirrors.kernel.org/sourceware/cygwin-archive/20221123'
+            $args += 'http://mirrors.kernel.org/sourceware/cygwin-archive/20221123'
           } else {
-            $site = 'http://mirrors.kernel.org/sourceware/cygwin/'
+            $args += 'http://mirrors.kernel.org/sourceware/cygwin/'
           }
         } else {
-          $site = '${{ inputs.site }}'
+          $sites = '${{ inputs.sites }}'
+          $site_list = $sites.Split('', [System.StringSplitOptions]::RemoveEmptyEntries)
+          $site_list = $site_list | % { $_.Trim() }
+          foreach ($site in $site_list) {
+            $args += '-s'
+            $args += $site
+          }
```

Comment:
It looks like this could be written more clearly by splitting into two parts,: setting $sites to the appopriate default if it's not supplied, and then iterating over the split value of $sites to add to args